test/e2e/on-request-error/isr/isr.test.ts

```diff
@@ -0,0 +1,90 @@
+import { nextTestSetup } from 'e2e-utils'
+import { retry, waitFor } from 'next-test-utils'
+import { getOutputLogJson } from '../_testing/utils'
+
+const outputLogPath = 'output-log.json'
+
+const isPPREnabledByDefault = process.env.__NEXT_EXPERIMENTAL_PPR === 'true'
+
+describe('on-request-error - isr', () => {
+  const { next, skipped, isNextDev } = nextTestSetup({
+    files: __dirname,
+    skipDeployment: true,
+  })
+
+  if (skipped) {
+    return
+  }
+
+  if (isNextDev) {
+    it('should skip in development mode', () => {
+      // This ISR test is only applicable for production mode
+    })
+    return
+  }
+
+  async function matchRevalidateReason(
+    errorMessage: string,
+    revalidateReason: string
+  ) {
+    await retry(async () => {
+      const json = await getOutputLogJson(next, outputLogPath)
+      expect(json[errorMessage]).toMatchObject({
+        context: {
+          revalidateReason,
+        },
+      })
+    })
+  }
+
+  describe('app router ISR', () => {
+    // TODO: support isr reason in PPR for app page
+    if (!isPPREnabledByDefault) {
+      it('should capture correct reason for stale errored page', async () => {
+        await next.fetch('/app/stale')
+        await waitFor(2 * 1000) // wait for revalidation
+        await next.fetch('/app/stale')
+
+        await matchRevalidateReason('app:stale', 'stale')
+      })
+
+      it('should capture correct reason for on-demand revalidated page', async () => {
+        await next.fetch('/app/on-demand')
+        await next.fetch('/api/revalidate-path?path=/app/on-demand')
+
+        await matchRevalidateReason('app:on-demand', 'on-demand')
+      })
+    }
+
+    it('should capture correct reason for build errored route', async () => {
+      await next.fetch('/app/route/stale')
+      await waitFor(2 * 1000) // wait for revalidation
+      await next.fetch('/app/route/stale')
+
+      await matchRevalidateReason('app:route:stale', 'stale')
+    })
+
+    it('should capture correct reason for on-demand revalidated route', async () => {
+      await next.fetch('/app/route/on-demand')
```

Comment:
for the on-demand revalidate tests, do we need to fetch the route itself or only the revalidation endpoint?